WooCommerce/Classes/Authentication/Application Password/ApplicationPasswordAuthorizationWebViewController.swift

```diff
@@ -157,20 +162,41 @@ private extension ApplicationPasswordAuthorizationWebViewController {
                 guard let url = try await viewModel.fetchAuthURL() else {
                     DDLogError("⛔️ No authorization URL found for application passwords")
                     analytics.track(.applicationPasswordAuthorizationURLNotAvailable)
-                    return showErrorAlert(message: Localization.applicationPasswordDisabled)
+                    navigateToErrorUI()
+                    return
                 }
                 loadAuthorizationPage(url: url)
             } catch {
                 DDLogError("⛔️ Error fetching authorization URL for application passwords \(error)")
                 analytics.track(.applicationPasswordAuthorizationURLFetchFailed, withError: error)
-                showErrorAlert(message: Localization.errorFetchingAuthURL, onRetry: { [weak self] in
-                    self?.fetchAuthorizationURL()
-                })
+                navigateToErrorUI()
```

Comment:
To be safe, we should rename `navigateToErrorUI` to `navigateToApplicationPasswordDisabledUI` and trigger it only when catching `Alamofire.AFError.requestAdaptationFailed(error: Networking.RequestAuthenticatorError.applicationPasswordNotAvailable)`

Reply:
I think then we should maintain the error alert for all other cases. This way, we can address any unexpected error that might arise other then the application password not being available